runtime/src/main/java/org/corfudb/runtime/view/BatchSequencerView.java

```diff
@@ -0,0 +1,252 @@
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.Data;
+import org.corfudb.protocols.wireprotocol.BatchTokenRequest;
+import org.corfudb.protocols.wireprotocol.BatchTokenResponse;
+import org.corfudb.protocols.wireprotocol.BatchTokenResponse.BackpointerToken;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.protocols.wireprotocol.TokenType;
+import org.corfudb.protocols.wireprotocol.TxResolutionInfo;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+
+public class BatchSequencerView extends SequencerView {
+
+    @Data
+    private static class BatchSequencerOperation {
+
+        final boolean next;
+
+        final TxResolutionInfo info;
+
+        final UUID[] streams;
+
+        final CompletableFuture<TokenResponse> completion = new CompletableFuture<>();
+
+        TokenResponse response = null;
+
+        void completeResponse() {
+            if (response == null) {
+                throw new UnrecoverableCorfuError("Attempted to complete an empty token");
+            }
+            completion.complete(response);
+        }
+    }
+
+    Semaphore s = new Semaphore(0);
+
+    final Deque<BatchSequencerOperation> queue = new ConcurrentLinkedDeque<>();
+
+    final ExecutorService batcher = Executors.newFixedThreadPool(1, new ThreadFactoryBuilder()                                                  .setNameFormat("sequencer-batcher-%d").build());
+
+    volatile long lastGlobal = Address.NEVER_READ;
+    volatile long lastEpoch = -1;
+
+    public BatchSequencerView(CorfuRuntime runtime) {
+        super(runtime);
+        batcher.submit(this::sequencerBatcher);
+    }
+
+    private void sequencerBatcher() {
+        // A list of completed operations
+        List<BatchSequencerOperation> readOps = new ArrayList<>();
+        List<BatchSequencerOperation> writeOps = new ArrayList<>();
+        List<BatchSequencerOperation> conditionalWriteOps = new ArrayList<>();
+        List<BatchSequencerOperation> abortedOps = new ArrayList<>();
+
+        Set<UUID> readStreams = new HashSet<>();
+        List<UUID[]> tokenRequests = new ArrayList<>();
+        List<TxResolutionInfo> conditionalTokenRequests = new ArrayList<>();
+        SetMultimap<UUID, ByteBuffer> conflictMap =
+            MultimapBuilder.hashKeys().hashSetValues().build();
+
+        while (!runtime.isShutdown()) {
+            readOps.clear();
+            writeOps.clear();
+            conditionalWriteOps.clear();
+            abortedOps.clear();
+
+            readStreams.clear();
+            tokenRequests.clear();
+            conditionalTokenRequests.clear();
+            conflictMap.clear();
+
+            try {
+                s.tryAcquire(runtime.getParameters().getBatchSequencerRequestCount()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "tryAcquire". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)